### DISCUSSION TITLE
How do you create an arbitray for a Map() object?

### DISCUSSION THREAD

**Discussion by dpnova** [dubzzz/fast-check#4992] [Q&A] [answered]
I have code that uses `.entries` and none of the current map-ish arbs support that.

New user here so maybe I've missed something!

Thanks

**Comment by gruhn** [comment, marked as answer]
If your keys are strings, you can use [`fc.dictionary`](https://fast-check.dev/docs/core-blocks/arbitraries/composites/object/#dictionary). For example
```typescript
const arbitraryMap = fc.dictionary(fc.string(), fc.nat())
    .map(dict => new Map(Object.entries(dict)))

console.log( fc.sample(arbitraryMap, 5) )
```
```typescript
Map(1) { 'CxbD!_' => 1135090401 },
Map(3) { 'q`' => 2113644272, ' x&~' => 27, '~}!t|z~y' => 25 },
Map(1) { '_#3' => 1738777444 },
Map(4) {
  '|call}&b #' => 9,
  '\\^!+L(' => 28,
  '' => 1752156948,
  'c' => 163999411
},
Map(7) {
  'I>U){' => 1984610647,
  'KEeeK' => 1524839195,
  'sAZ' => 937344058,
  '' => 529998270,
  'Gd2#<=ZHr}' => 1135358149,
  'a;kS4rzo' => 1594872950,
  ':(' => 641853712
}
```

**Reply by dpnova** [reply]
thanks @gruhn - I thought I was doing that, the type error for me seems to be on a nested field. In the screenshot you can see the `ff` function the line above the error where I have a function that takes a single instance of the nested type and the compiler is ok with that, it's just not ok when that appears as a value of a nested map.

![image](https://github.com/dubzzz/fast-check/assets/229943/52efdc9c-e2af-40b7-a196-99206d77908a)
![image](https://github.com/dubzzz/fast-check/assets/229943/f6d8e0c3-d895-4850-822d-ad0d01a9ea27)

![image](https://github.com/dubzzz/fast-check/assets/229943/a5b157a2-3d3e-467c-b0ef-c6548dc7875a)

**Reply by dpnova** [reply]
![image](https://github.com/dubzzz/fast-check/assets/229943/ab7830fb-ae8f-4ca5-a711-1354215d095f)

This counter example shows an object for `items` not a map. Arbitrary below - I tried forcing `minKeys` to stop it from using `{}` for items - but it still inserts an object, not a map.

![image](https://github.com/dubzzz/fast-check/assets/229943/7368704d-cf0e-49e3-b772-46eb92265a6a)

**Reply by dpnova** [reply]
Oh I finally realised what your `map` was doing in your example!

Thanks! This was very hard to google for
